app/scripts/metamask-controller.js

```diff
@@ -5083,6 +5085,10 @@ export default class MetamaskController extends EventEmitter {
     engine.push(createLoggerMiddleware({ origin }));
     engine.push(this.permissionLogController.createMiddleware());
 
+    if (origin === BaseUrl.Portfolio) {
```

Comment:
Rather than coupling this to a specific origin, should we make it a little more generic and extensible via an array constant such as `TRANSACTION_VERIFICATION_ORIGINS`?

Reply:
We could, but Portfolio is the only affected product, and there are no other supported products that require this guard (yet) that wouldn't already be a part of Portfolio. If there are services outside of portfolio that would warrant this change I'm for it.

Reply:
I have modified the code so that it still targets `origin === BaseUrl.Portfolio`, but that any experience could easily add an "experience to verify" with "trusted signers"